libexec/pyenv-version-file-read

```diff
@@ -6,16 +6,18 @@ set -e
 VERSION_FILE="$1"
 
 if [ -s "$VERSION_FILE" ]; then
-  # Read the first non-whitespace word from the specified version file.
-  # Be careful not to load it whole in case there's something crazy in it.
+  # Read the first word from the specified version file. Avoid reading it whole.
   IFS="${IFS}"$'\r'
-  sep=
-  while read -n 1024 -r version _ || [[ $version ]]; do
-      [[ -z $version || $version == \#* ]] && continue
-      printf "%s%s" "$sep" "$version"
-      sep=:
-  done <"$VERSION_FILE"
-  [[ $sep ]] && { echo; exit; }
+  read -n 1024 -d "" -r version _ <"$VERSION_FILE" || :
```

Comment:
The removal of the loop will disallow parsing multiple versions in the file. Rbenv doesn't support that apparently.

Tests are also missing.

Reply:
I amended my work again, this time I took the time to add some new unit tests as well :)